### PR TITLE
Add documentation to enable clients using Bot as API to send feedback to Haptik API

### PR DESCRIPTION
**Pull request by haptikrajasashtikar**
- Add guidelines for clients using Bot as API to send feedback to Haptik API

CIA-6688

### CODE REVIEW

File: docs/webhooks/send-feedback-to-haptik.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
+title: How to send feedback to Haptik
+---
+
+## Log Feedback to Haptik via Rest Api
```
Comment (tejas-haptik) [inline docs/webhooks/send-feedback-to-haptik.md:5]: `REST API`

File: docs/webhooks/send-feedback-to-haptik.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+The Log Feedback API allows you to send feedback via a `POST` request to the Haptik Platform. The URL for logging feedback is generated on the Haptik Platform Dashboard.
+
+### URL
+
+`https://staging-messenger.haptikapi.com/v1.0/log_feedback_from_user/`
```
Comment (tejas-haptik) [inline docs/webhooks/send-feedback-to-haptik.md:11]: Let's remove the base URL and add a place holder

File: docs/webhooks/send-feedback-to-haptik.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  "auth_id": <AUTH_ID>,
+  "business_id": <BUSINESS_ID>,
+  "query_resolved": true,
+  "query_rating": 5,
+  "feedback": "The conversation was very human like"
+}
+```
+
+- auth_id (required) - This is an alphanumeric User identifier from your system
+- business_id (required) - This is a numeric identifier for channel/queue that you wish to register the feedback on.
```
Comment (tejas-haptik) [inline docs/webhooks/send-feedback-to-haptik.md:38]: `channel/queue` let's check once what are we using across different pages.

@medhaanand can you please help here to define what is naming convention we are following across docs? Is it `business`?

File: docs/webhooks/send-feedback-to-haptik.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+- query_resolved (required) - This is a boolean indicator of whether the user's query was resolved or not.
+- query_rating (required) - This is a numeric indicator on the scale of 5 of how satisfied was the user with the conversation.
```
Comment (tejas-haptik) [inline docs/webhooks/send-feedback-to-haptik.md:40]: `scale of 0 to 5, to tell how satisfied...`

File: docs/webhooks/send-feedback-to-haptik.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+```json
+{
+  "error_message": "invalid authorization details"
+}
+```
+
+## Sample CURL command
+
+```
+curl -X POST 'https://staging-messenger.haptikapi.com/v1.0/log_feedback_from_user/' \
```
Comment (tejas-haptik) [inline docs/webhooks/send-feedback-to-haptik.md:91]: let's put a placeholder for base URL

File: docs/webhooks/send-feedback-to-haptik.md (hunk 2 above)
Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md:11]: This URL appears across message and file upload pages as well.
Do we remove from those as well?

File: docs/webhooks/send-feedback-to-haptik.md (hunk 3 above)
Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md:38]: This is the same description from the log message to haptik api page

File: docs/webhooks/send-feedback-to-haptik.md (hunk 4 above)
Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md:40]: Scale of 1 - 5, rest I will make the changes

File: docs/webhooks/send-feedback-to-haptik.md (hunk 5 above)
Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md:91]: Again this appears across log message and file upload, do we change there as well?

File: docs/webhooks/send-feedback-to-haptik.md (hunk 3 above)
Comment (medhaanand) [reply docs/webhooks/send-feedback-to-haptik.md:38]: Waiting for @RandiveSushant  to confirm about when should this docs be released. Will reply on this thread.

File: docs/webhooks/send-feedback-to-haptik.md (hunk 2 above)
Comment (tejas-haptik) [reply docs/webhooks/send-feedback-to-haptik.md:11]: Would be great if we can remove them as well.

File: docs/webhooks/send-feedback-to-haptik.md (hunk 3 above)
Comment (tejas-haptik) [reply docs/webhooks/send-feedback-to-haptik.md:38]: @haptikrajasashtikar Cool. Let's not merge this PR until the changes are live. If possible let's `Don't Merge` in PR Title.

@medhaanand We'll release this doc once the changes are tested and live.

File: docs/webhooks/send-feedback-to-haptik.md (hunk 4 above)
Comment (tejas-haptik) [reply docs/webhooks/send-feedback-to-haptik.md:40]: Cool 👍

File: docs/webhooks/send-feedback-to-haptik.md (hunk 5 above)
Comment (tejas-haptik) [reply docs/webhooks/send-feedback-to-haptik.md:91]: Yes please. Let's update this wherever you find it

File: docs/webhooks/send-feedback-to-haptik.md (hunk 2 above)
Comment (medhaanand) [reply docs/webhooks/send-feedback-to-haptik.md:11]: Yes we should remove it from the other two pages too

File: docs/webhooks/send-feedback-to-haptik.md [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  "query_rating": 5,
+  "feedback": "The conversation was very human like"
+}
+```
+
+- auth_id (required) - This is an alphanumeric User identifier from your system
+- business_id (required) - This is a numeric identifier for channel/queue that you wish to register the feedback on.
+- query_resolved (required) - This is a boolean indicator of whether the user's query was resolved or not.
+- query_rating (required) - This is a numeric indicator on the scale of 5 of how satisfied was the user with the conversation.
+- feedback (option) - This is a string field, allowing user to provide written feedback pointing to specifics of their experience.
```
Comment (medhaanand) [inline docs/webhooks/send-feedback-to-haptik.md:41]:
```suggestion
- feedback (optional) - This is a string field, allowing user to provide written feedback pointing to specifics of their experience. 
```

File: docs/webhooks/send-feedback-to-haptik.md [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  "business_id": <BUSINESS_ID>,
+  "query_resolved": true,
+  "query_rating": 5,
+  "feedback": "The conversation was very human like"
+}
+```
+
+- auth_id (required) - This is an alphanumeric User identifier from your system
+- business_id (required) - This is a numeric identifier for channel/queue that you wish to register the feedback on.
+- query_resolved (required) - This is a boolean indicator of whether the user's query was resolved or not.
```
Comment (medhaanand) [inline docs/webhooks/send-feedback-to-haptik.md:39]: how will this be sent? Dont we have only query rating and feedback?

File: docs/webhooks/send-feedback-to-haptik.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+
+```json
+{
+  "success": true
+}
+```
+
+### Error Response
+
+To have a fair judgement of a conversation and the agent, a user is allowed to submit feedback only once in 1 hour. If the user tries to submit the feedback again within 1 hour, an error will be raised
```
Comment (medhaanand) [inline docs/webhooks/send-feedback-to-haptik.md:55]:
```suggestion
To have a fair judgement of a conversation or an agent, a user is allowed to submit feedback only once in 1 hour. If the user tries to submit the feedback again within 1 hour, an error will be raised
```

File: docs/webhooks/send-feedback-to-haptik.md (hunk 3 above)
Comment (medhaanand) [reply docs/webhooks/send-feedback-to-haptik.md:38]: We will take this docs live now, @RandiveSushant and I are also working on making it accessible as a template. I will put the docs for that later.

Comment (tejas-haptik) [reply docs/webhooks/send-feedback-to-haptik.md:38]: We are taking it live now? @RandiveSushant @medhaanand 

What do we mean by this? This feature is not yet released.

File: docs/webhooks/send-feedback-to-haptik.md (hunk 7 above)
Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md:39]: As in?
So in Haptik SDK, how it works is, if the query rating is below 3, query_resolved is sent as False.
Clients can take a call, when do they decide that the query_resolved is True or False.

Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md:39]: But query_resolved is checked while logging the message in Haptik API, if query was resolve we log a positive message, else we log a negative message and the ECM metrics of the Agent also get affected accordingly.

File: docs/webhooks/send-feedback-to-haptik.md (hunk 4 above)
Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md:40]: Done

File: docs/webhooks/send-feedback-to-haptik.md (hunk 5 above)
Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md]: Done

File: docs/webhooks/send-feedback-to-haptik.md (hunk 3 above)
Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md:38]: Done

Comment (medhaanand) [reply docs/webhooks/send-feedback-to-haptik.md:38]: Oh I thought its already released. We will take it live once its released then.

File: docs/webhooks/send-feedback-to-haptik.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
+---
+title: How to send feedback to Haptik
+---
+
+## Log Feedback to Haptik via REST Api
```
Comment (medhaanand) [inline docs/webhooks/send-feedback-to-haptik.md:5]:
```suggestion
## Log Feedback to Haptik via REST API
```

File: docs/webhooks/send-message-to-haptik.md [hunk 10]
```diff
@@ -2,13 +2,13 @@
 title: How to send message to Haptik
 ---
 
-## Log Message to Haptik via Rest Api
+## Log Message to Haptik via REST Api
```
Comment (medhaanand) [inline docs/webhooks/send-message-to-haptik.md:5]:
```suggestion
## Log Message to Haptik via REST API
```

File: docs/webhooks/send-feedback-to-haptik.md (hunk 7 above)
Comment (medhaanand) [reply docs/webhooks/send-feedback-to-haptik.md:39]:
```suggestion
- query_resolved (required) - This is a boolean indicator of whether the user's query was resolved or not. This helps us bucket conversations into positive or negative categories.
```

File: docs/webhooks/send-feedback-to-haptik.md (hunk 1 above)
Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md]: Done

File: docs/webhooks/send-feedback-to-haptik.md (hunk 2 above)
Comment (haptikrajasashtikar) [reply docs/webhooks/send-feedback-to-haptik.md]: Done

**Review by medhaanand — APPROVED** [review]

**Pull request merged**